Modules/Filtering/ImageGrid/include/itkResampleImageFilter.h

```diff
@@ -111,16 +111,16 @@ class ITK_TEMPLATE_EXPORT ResampleImageFilter : public ImageToImageFilter<TInput
   itkTypeMacro(ResampleImageFilter, ImageToImageFilter);
 
   /** Number of dimensions. */
-  static constexpr unsigned int ImageDimension = TOutputImage::ImageDimension;
+  static constexpr unsigned int OutputImageDimension = TOutputImage::ImageDimension;
   static constexpr unsigned int InputImageDimension = TInputImage::ImageDimension;
 
-  /** base type for images of the current ImageDimension */
-  using ImageBaseType = ImageBase<Self::ImageDimension>;
+  /** base type for images of the current OutputImageDimension */
+  using ImageBaseType = ImageBase<Self::OutputImageDimension>;
 
   /**
    *  Transform type alias.
    */
-  using TransformType = Transform<TTransformPrecisionType, Self::ImageDimension, Self::ImageDimension>;
+  using TransformType = Transform<TTransformPrecisionType, Self::OutputImageDimension, Self::InputImageDimension>;
```

Comment:
In the resampling process we fill in values at each point in output image by finding corresponding points in input image. So the transform is used to map each output point to corresponding point in input space, therefore output dimensions come first.

Reply:
> transform is used to map each output point to corresponding point in input space

ITK's infrastructure already does this, so you might be double-inverting it here.

Reply:
Yes, ResampleImageFilter already does this.. see for example this [line](https://github.com/InsightSoftwareConsortium/ITK/blob/f15a14b3cacdc04eb20edaa225a730769cd89612/Modules/Filtering/ImageGrid/include/itkResampleImageFilter.hxx#L377). where output point is transformed to input point. So the transform has to map OutputSpace -> InputSpace.

Reply:
This is clearly a complicated issue. Reading over the class level documentation, it does not clearly state that the transform maps from the OutputImageSpace to the InputImageSpace. I believe this is outside of the scope of this PR, but as this one of the most heavily used classes and it's concepts are a precursor to understanding registration. It could use improvement.